be/src/exec/hash_join_components.cpp

```diff
@@ -85,6 +100,241 @@ void HashJoinProber::attach(HashJoinBuilder* builder, const HashJoinProbeMetrics
     _impl = builder->create_prober();
 }
 
+class PartitionChunkChannel {
+public:
+    PartitionChunkChannel(MemTracker* tracker) : _tracker(tracker) {}
+    bool processing() const { return _processing; }
+    void set_processing(bool processing) { _processing = processing; }
+
+    ChunkPtr pull() {
+        auto chunk = std::move(_chunks.front());
+        _tracker->release(chunk->memory_usage());
+        _chunks.pop_front();
+        return chunk;
+    }
+
+    void push(ChunkPtr&& chunk) {
+        _tracker->consume(chunk->memory_usage());
+        _chunks.emplace_back(std::move(chunk));
+    }
+
+    const ChunkPtr& back() { return _chunks.back(); }
+
+    bool is_full() const {
+        return _chunks.size() >= 4 || _tracker->consumption() > config::local_exchange_buffer_mem_limit_per_driver;
```

Comment:
Why use the config: `local_exchange_buffer_mem_limit_per_driver` ?

Reply:
add a new config partition_hash_join_probe_limit_size